--- /arch/arm/mach-at91/board-sam9260ek.c	2010-11-22 20:03:49.000000000 +0100
+++ /board-mmnet1000.c	2010-12-16 13:28:01.000000000 +0100
@@ -54,7 +54,7 @@
 	/* Initialize processor: 18.432 MHz crystal */
 	at91sam9260_initialize(18432000);
 
-	/* DBGU on ttyS0. (Rx & Tx only) */
+	/* DGBU on ttyS0. (Rx & Tx only) */
 	at91_register_uart(0, 0, 0);
 
 	/* USART0 on ttyS1. (Rx, Tx, CTS, RTS, DTR, DSR, DCD, RI) */
@@ -90,6 +90,16 @@
 	.pullup_pin	= 0,		/* pull-up driven by UDC */
 };
 
+/*
+ * Compact Flash (via Expansion Connector)
+ */
+static struct at91_cf_data __initdata ek_cf_data = {
+	// .irq_pin	= ... user defined
+	// .det_pin	= ... user defined
+	// .vcc_pin	= ... user defined
+	// .rst_pin	= ... user defined
+	.chipselect	= 4,
+};
 
 /*
  * Audio
@@ -158,7 +168,7 @@
  */
 static struct at91_eth_data __initdata ek_macb_data = {
 	.phy_irq_pin	= AT91_PIN_PA7,
-	.is_rmii	= 1,
+	.is_rmii	= 0,
 };
 
 
@@ -167,12 +177,27 @@
  */
 static struct mtd_partition __initdata ek_nand_partition[] = {
 	{
-		.name	= "Partition 1",
+		.name	= "bootstrap",
 		.offset	= 0,
 		.size	= SZ_256K,
 	},
 	{
-		.name	= "Partition 2",
+		.name	= "u-boot",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= SZ_256K,
+	},
+	{
+		.name	= "u-boot environment",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= SZ_1M + SZ_512K,
+	},
+	{
+		.name	= "kernel",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= SZ_8M - SZ_2M,
+	},
+	{
+		.name	= "filesystems",
 		.offset	= MTDPART_OFS_NXTBLK,
 		.size	= MTDPART_SIZ_FULL,
 	},
@@ -197,7 +222,7 @@
 	.bus_width_16	= 0,
 #endif
 };
-
+/*
 static struct sam9_smc_config __initdata ek_nand_smc_config = {
 	.ncs_read_setup		= 0,
 	.nrd_setup		= 1,
@@ -218,17 +243,18 @@
 
 static void __init ek_add_device_nand(void)
 {
-	/* setup bus-width (8 or 16) */
+	// setup bus-width (8 or 16) 
 	if (ek_nand_data.bus_width_16)
 		ek_nand_smc_config.mode |= AT91_SMC_DBW_16;
 	else
 		ek_nand_smc_config.mode |= AT91_SMC_DBW_8;
 
-	/* configure chip-select 3 (NAND) */
+	// configure chip-select 3 (NAND) 
 	sam9_smc_configure(3, &ek_nand_smc_config);
 
 	at91_add_device_nand(&ek_nand_data);
 }
+*/
 
 
 /*
@@ -254,12 +280,14 @@
 		.default_trigger	= "none",
 	},
 	{	/* "power" led, yellow */
-		.name			= "ds1",
-		.gpio			= AT91_PIN_PA9,
+		.name			= "usr",
+		.gpio			= AT91_PIN_PC15,
+		.active_low		= 1,
 		.default_trigger	= "heartbeat",
 	}
 };
 
+
 /*
  * I2C devices
  */
@@ -338,13 +366,16 @@
 	/* SPI */
 	at91_add_device_spi(ek_spi_devices, ARRAY_SIZE(ek_spi_devices));
 	/* NAND */
-	ek_add_device_nand();
+	//ek_add_device_nand();
+	at91_add_device_nand(&ek_nand_data);
 	/* Ethernet */
 	at91_add_device_eth(&ek_macb_data);
 	/* MMC */
 	at91_add_device_mmc(0, &ek_mmc_data);
 	/* I2C */
 	at91_add_device_i2c(ek_i2c_devices, ARRAY_SIZE(ek_i2c_devices));
+	/* Compact Flash */
+	at91_add_device_cf(&ek_cf_data);
 	/* SSC (to AT73C213) */
 	at73c213_set_clk(&at73c213_data);
 	at91_add_device_ssc(AT91SAM9260_ID_SSC, ATMEL_SSC_TX);
@@ -352,10 +383,13 @@
 	at91_gpio_leds(ek_leds, ARRAY_SIZE(ek_leds));
 	/* Push Buttons */
 	ek_add_device_buttons();
+	/* shutdown controller, wakeup button (5 msec low) */
+	at91_sys_write(AT91_SHDW_MR, AT91_SHDW_CPTWK0_(10) | AT91_SHDW_WKMODE0_LOW
+				| AT91_SHDW_RTTWKEN);
 }
 
-MACHINE_START(AT91SAM9260EK, "Atmel AT91SAM9260-EK")
-	/* Maintainer: Atmel */
+MACHINE_START(AT91SAM9260EK, "Apex Systems Easyplus")
+	/* Maintainer: Sagaert Johan Apex */
 	.phys_io	= AT91_BASE_SYS,
 	.io_pg_offst	= (AT91_VA_BASE_SYS >> 18) & 0xfffc,
 	.boot_params	= AT91_SDRAM_BASE + 0x100,
